biscuit-auth/src/format/mod.rs

```diff
@@ -291,7 +294,21 @@ impl SerializedBiscuit {
         next_keypair: &KeyPair,
         authority: &Block,
     ) -> Result<Self, error::Token> {
-        Self::new_inner(root_key_id, root_keypair, next_keypair, authority, 0)
+        let authority_signature_version = if authority.version >= DATALOG_3_3 {
+            DATALOG_3_3_SIGNATURE_VERSION
+        } else {
+            match (root_keypair, next_keypair) {
+                (KeyPair::Ed25519(_), KeyPair::Ed25519(_)) => 0,
+                _ => NON_ED25519_SIGNATURE_VERSION,
+            }
+        };
```

Comment:
the version decision should be moved to another function that can be reused for each block

Reply:
the decision is a bit different based on each case (creating a biscuit does not have previous `crypto::Block` available, `append_serialized` does not have a `Block` available either), so the common function will be a bit less readable

Reply:
i have done it in a separate commit. the function itself is a bit harder to read and we’re forced to use explicit iterators, but it’s indeed better to have the logic in a single place. as a bonus i was able to write tests instead of just relying on samples